ppa/archive/views.py

```diff
@@ -81,7 +87,47 @@ def get_queryset(self, **kwargs):
         # sort so work is first, then by page order
         filter_q.append('{!collapse field=srcid sort="order asc"}')
 
-        self.solrq = PagedSolrQuery({
+        range_opts = {
+            'facet.range': self.form.range_facets
+        }
+        for range_facet in self.form.range_facets:
+            # range filter requested in search options
+            start = end = None
+            # if start or end is specified on the form, add a filter query
+            if range_facet in search_opts and search_opts[range_facet]:
+                start, end = search_opts[range_facet].split('-')
+                range_filter = '[%s TO %s]' % (start or '*', end or '*')
+                filter_q.append('%s:%s' % (range_facet, range_filter))
+
+            # get minimum and maximum pub date values from the db
+            pubmin, pubmax = self.form.pub_date_minmax()
+
+            start = int(start) if start else pubmin
+            end = int(end) if end else pubmax
+
+            # fallback logic for when start and end are not set
+            # (should only happen when no content is in the database)
+            if not start or end:
```

Comment:
how can we get here after lines 105-106? if start/end are `None` won't they be set above?

Reply:
is this supposed to be `if not pubmin or pubmax`, etc?

Reply:
Good point.  Just simplified to 
```
start = int(start) if start else pubmin or 0
end = int(end) if end else pubmax or 1922
```